Mandrel 21.2 branch fails to build with 11.0.14-dev (Inet6Address missing cached_scope_id)

Building mandrel from the 21.2 branch (21.3 branch likely has the same issue) with OpenJDK 11.0.14 results in:

```
DEBUG [build] Launcher line AFTER: exec "${location}/../../../bin/java" -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -Dorg.graalvm.version="21.2.0.3-dev" -Dorg.graalvm.config="Mandrel Distribution" --upgrade-module-path ${location}/../../jvmci/graal.jar --upgrade-module-path ${location}/../../jvmci/graal.jar --add-modules org.graalvm.truffle,org.graalvm.sdk --module-path ${location}/../../truffle/truffle-api.jar:${location}/../../jvmci/graal-sdk.jar "${jvm_args[@]}" ${app_path_arg} "${cp_or_mp}" ${main_class} "${launcher_args[@]}"
DEBUG [build] Build native agents...
DEBUG [OperatingSystem] Execute [/disk/graal/upstream-sources/graalvm/mandrel-build/lib/svm/bin/native-image, --macro:native-image-agent-library] in /disk/graal/upstream-sources/mandrel-packaging
[libnative-image-agent:28605]    classlist:   1,549.78 ms,  0.96 GB
--initialize-at-build-time without arguments has been deprecated and will be removed in GraalVM 22.0.
[libnative-image-agent:28605]        (cap):     863.06 ms,  0.96 GB
[libnative-image-agent:28605]        setup:   2,652.13 ms,  0.96 GB
[libnative-image-agent:28605]     analysis:  11,748.52 ms,  1.76 GB
Fatal error:com.oracle.svm.util.ReflectionUtil$ReflectionUtilError: java.lang.NoSuchFieldException: cached_scope_id
	at com.oracle.svm.util.ReflectionUtil.lookupField(ReflectionUtil.java:114)
	at com.oracle.svm.util.ReflectionUtil.lookupField(ReflectionUtil.java:101)
	at com.oracle.svm.core.jdk.JNIRegistrationUtil.fields(JNIRegistrationUtil.java:104)
	at com.oracle.svm.hosted.jdk.JNIRegistrationJavaNet.registerInitInetAddressIDs(JNIRegistrationJavaNet.java:194)
	at com.oracle.svm.hosted.jdk.JNIRegistrationJavaNet.registerNetworkInterfaceInit(JNIRegistrationJavaNet.java:220)
	at com.oracle.svm.hosted.ReachabilityHandlerFeature.duringAnalysis(ReachabilityHandlerFeature.java:119)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$12(NativeImageGenerator.java:730)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:730)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:532)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:491)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:380)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:543)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:119)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:573)
Caused by: java.lang.NoSuchFieldException: cached_scope_id
	at java.base/java.lang.Class.getDeclaredField(Class.java:2411)
	at com.oracle.svm.util.ReflectionUtil.lookupField(ReflectionUtil.java:106)
	... 14 more
[libnative-image-agent:28605]      [total]:  16,161.01 ms,  1.76 GB
# Printing build artifacts to: /disk/graal/upstream-sources/graalvm/mandrel-build/lib/svm/macros/native-image-agent-library/../../../libnative-image-agent.build_artifacts.txt
```

Root cause seems to be [JDK-8216417](https://bugs.openjdk.java.net/browse/JDK-8216417) which is new in 11.0.14-dev

--- a/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationJavaNet.java
+++ b/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationJavaNet.java
@@ -24,6 +24,7 @@
  */
 package com.oracle.svm.hosted.jdk;
 
+import java.lang.reflect.Field;
 import java.net.DatagramPacket;
 import java.net.InetAddress;
 import java.net.Proxy;
@@ -40,6 +41,7 @@ import com.oracle.svm.core.jdk.JNIRegistrationUtil;
 import com.oracle.svm.core.jni.JNIRuntimeAccess;
 import com.oracle.svm.core.util.VMError;
 import com.oracle.svm.hosted.FeatureImpl.DuringAnalysisAccessImpl;
+import com.oracle.svm.util.ReflectionUtil;
 
 /**
  * Registration of classes, methods, and fields accessed via JNI by C code of the JDK.
@@ -190,8 +192,13 @@ class JNIRegistrationJavaNet extends JNIRegistrationUtil implements Feature {
         /* Java_java_net_Inet6Address_init */
         JNIRuntimeAccess.register(constructor(a, "java.net.Inet6Address"));
         JNIRuntimeAccess.register(fields(a, "java.net.Inet6Address", "holder6"));
-        if (JavaVersionUtil.JAVA_SPEC < 13) {
-            JNIRuntimeAccess.register(fields(a, "java.net.Inet6Address", "cached_scope_id"));
+        if (JavaVersionUtil.JAVA_SPEC < 13) { // JDK-8216417
+            Class<?> c = clazz(a, "java.net.Inet6Address");
+            boolean optional = JavaVersionUtil.JAVA_SPEC == 11; // JDK-8269385
+            Field f = ReflectionUtil.lookupField(optional, c, "cached_scope_id");
+            if (f != null) {
+                JNIRuntimeAccess.register(f);
+            }
         }
         JNIRuntimeAccess.register(fields(a, "java.net.Inet6Address$Inet6AddressHolder", "ipaddress", "scope_id", "scope_id_set", "scope_ifname"));
     }

